packages/webview_flutter/example/lib/main.dart

```diff
@@ -55,26 +55,66 @@ class WebViewExample extends StatelessWidget {
   }
 }
 
+enum MenuOptions {
+  evaluateJavaScript,
+  toast,
+}
+
 class SampleMenu extends StatelessWidget {
-  const SampleMenu();
+  SampleMenu(this.controller);
+  final Completer<WebViewController> controller;
 
   @override
   Widget build(BuildContext context) {
-    return PopupMenuButton<String>(
-      onSelected: (String value) {
-        Scaffold.of(context)
-            .showSnackBar(SnackBar(content: Text('You selected: $value')));
+    return FutureBuilder<WebViewController>(
+      future: controller.future,
+      builder:
+          (BuildContext context, AsyncSnapshot<WebViewController> controller) {
+        return PopupMenuButton<MenuOptions>(
+          onSelected: (MenuOptions value) {
+            switch (value) {
+              case MenuOptions.evaluateJavaScript:
+                controller.data
```

Comment:
nit: this can be cleaner if you extract it to an async method and use async/await.